test/Dialect/HW/basic.mlir

```diff
@@ -144,4 +144,39 @@ hw.module @signed_arrays(%arg0: si8) -> (out: !hw.array<2xsi8>) {
 // CHECK-LABEL: hw.module @argRenames
 // CHECK-SAME: attributes {argNames = [""]}
 hw.module @argRenames(%arg1: i32) attributes {argNames = [""]} {
-}
\ No newline at end of file
+}
+
+
+// CHECK-LABEL: OutputFileAttr
+hw.module @OutputFileAttr() -> () attributes {tests = [
+  // CHECK-SAME: #hw.output_file<"", "test0.sv">
+  #hw.output_file<"", "test0.sv">,
+
+  // CHECK-SAME: #hw.output_file<"dir", "test1.sv">,
+  #hw.output_file<"dir", "test1.sv">,
+
+  // CHECK-SAME: #hw.output_file<"dir", "test2.sv">
+  #hw.output_file<"", "dir/test2.sv">,
+
+  // CHECK-SAME: #hw.output_file<"/dir", "test3.sv">
+  #hw.output_file<"/dir/", "test3.sv">,
+
+  // CHECK-SAME: #hw.output_file<"/dir", "test4.sv">
+  #hw.output_file<"x", "/dir/test4.sv">,
+
+  // CHECK-SAME: #hw.output_file<"/dir", "test5.sv">
+  #hw.output_file<"", "/dir/test5.sv">,
+
+  // CHECK-SAME: #hw.output_file<"dir", "test6.sv">
+  #hw.output_file<"./x/x/./../..", "dir/./x/../test6.sv">,
+
+  // CHECK-SAME: #hw.output_file<"/", "">
+  #hw.output_file<"/", "">,
+
+  // CHECK-SAME: #hw.output_file<"/", "">
+  #hw.output_file<".", "/">,
+
+  // CHECK-SAME: #hw.output_file<"", "test.sv", excludeFromFileList, includeReplicatedOps>
+  #hw.output_file<"", "test.sv", excludeFromFileList, includeReplicatedOps>
```

Comment:
This PR is not done yet, but I wanted to open the doors for some syntax bikeshedding.  Right now the `OutputFileAttr` stores the directory and the filename as separate fields, and it always prints both, and the bool attributes are printed as keywords.  How does this syntax look?  Should the directory be omitted if empty?

Reply:
Also it's probably a little weird for the MLIR attribute parser to automagically normalize the file paths, but it helps for writing tests for the file normalizing code.

Reply:
I agree that having the parser normalize is a bit weird.  Instead of storing the fields separately, is there any downside to storing them together?

Reply:
I think we can do it, we just have to be very careful about keeping a final `/` on the path if its a directory.

Reply:
IR's are good for that!  That also allows you to have some nice "isDirectory()" methods etc.